packages/blockchain-wallet-v4-frontend/src/data/middleware/webSocket/socketd/sagas.js

```diff
@@ -0,0 +1,284 @@
+        send(
+          JSON.stringify({
+            command: 'subscribe',
+            entity: 'account',
+            coin: 'btc',
+            param: { address }
+          })
+        )
+        send(
+          JSON.stringify({
+            command: 'subscribe',
+            entity: 'account',
+            coin: 'bch',
+            param: { address }
+          })
+        )
+      })
+
+      // 3. subscribe to xpubs
+      walletContext.xpubs.forEach(xpub => {
+        send(
+          JSON.stringify({
+            command: 'subscribe',
+            entity: 'xpub',
+            coin: 'btc',
+            param: { address: xpub }
+          })
+        )
+        send(
+          JSON.stringify({
+            command: 'subscribe',
+            entity: 'xpub',
+            coin: 'bch',
+            param: { address: xpub }
+          })
+        )
+      })
+
+      // 4. subscribe to lockbox xpubs
+      const { data: btcLockboxContext } = yield select(
+        selectors.core.kvStore.lockbox.getLockboxBtcContext
+      )
+      btcLockboxContext.forEach(xpub =>
+        send(
+          JSON.stringify({
+            command: 'subscribe',
+            entity: 'xpub',
+            coin: 'btc',
+            param: { address: xpub }
+          })
+        )
+      )
+      const { data: bchLockboxContext } = yield select(
+        selectors.core.kvStore.lockbox.getLockboxBchContext
+      )
+      bchLockboxContext.forEach(xpub =>
+        send(
+          JSON.stringify({
+            command: 'subscribe',
+            entity: 'xpub',
+            coin: 'bch',
+            param: { address: xpub }
+          })
+        )
+      )
+
+      // 5. subscribe to ethereum addresses
+      const ethWalletContext = yield select(selectors.core.data.eth.getContext)
+      const { data: ethLockboxContext } = yield select(
+        selectors.core.kvStore.lockbox.getLockboxEthContext
+      )
+      const ethAddresses = concat(ethWalletContext, ethLockboxContext)
+      ethAddresses.forEach(address => {
+        send(
+          JSON.stringify({
+            command: 'subscribe',
+            entity: 'account',
+            coin: 'eth',
+            param: { address }
+          })
+        )
+      })
+    } catch (e) {
+      yield put(
+        actions.logs.logErrorMessage(
+          'middleware/webSocket/socketd/sagas',
+          'onOpen',
+          e.message
+        )
+      )
+    }
+  }
+
+  const onMessage = function * (action) {
+    const message = prop('payload', action)
+
+    try {
+      switch (message.coin) {
+        case 'btc':
+          if (header(message)) {
+            const header = prop('header', message)
+            yield put(
+              actions.core.data.btc.setBtcLatestBlock(
+                header.blockIndex,
+                header.hash,
+                header.height,
+                header.time
+              )
+            )
+          } else if (bitcoinTransaction(message)) {
```

Comment:
prefer `btc` over `bitcoin`

Reply:
I was using bitcoin because the condition is for both `btc` and `bch`. Is there a good shorthand for btc/bch grouping?